OpenRA.Game/Network/Order.cs

```diff
@@ -227,6 +228,40 @@ public static Order Deserialize(World world, BinaryReader r)
 			}
 		}
 
+		public static Order RefreshOrder(Order order)
+		{
+			if (order.Type != OrderType.Fields)
+				return order;
+
+			if (order.Subject != null && !order.Subject.IsInWorld)
```

Comment:
In Deserialise it's allowed for the Subject to be null when the order doesn't contain the `OrderFields.Subject` field

Reply:
Nope, see Order.cs, line 197. On Deserialise, if `OrderFields.Subject` is not null but the subject cannot be found in world (or `subject == null`), the order will be dropped.

It is also the direct causing for https://github.com/OpenRA/OpenRA/issues/20478

Reply:
There is a little bit misunderstand in my previous comment, you mean when order doesn't contain the OrderFields.Subject field, right?

However, for an order, if its `subject != null`, when serialising it will have a `OrderFields.Subject` tag. On the other clients that recieve the serialised order, if we have `OrderFields.Subject` tag but the subject cannot be found in world (or subject == null), the order will be dropped, which is also the direct causing for https://github.com/OpenRA/OpenRA/issues/20478

Reply:
so as for the local orders, if we find `subject != null`, that means it will send with `OrderFields.Subject` tag, we have to process it like Order.cs, line 197: if it cannot be found in world, we drop it.